addons/fuse-proxy/pkg/server/server.go

```diff
@@ -0,0 +1,222 @@
+		log.Errorf("Failed to marshal response: %v", err)
+	}
+	if err := mfcputil.SendMsg(unixConn, fd, b); err != nil {
+		log.Errorf("Failed to send response: %v", err)
+		return
+	}
+}
+
+func (s *Server) handleFusermount(req *common.Request, nsFd int) (int, error) {
+	nsPath := fmt.Sprintf("/proc/self/fd/%d", nsFd)
+	// Ensure /dev/fuse exists in the container's mnt namespace
+	if err := s.ensureFuseDevice(nsPath); err != nil {
+		return 0, fmt.Errorf("failed to ensure /dev/fuse exists: %v", err)
+	}
+	// Create socket pair for FD passing
+	socks, err := createSocketPair()
+	if err != nil {
+		return 0, fmt.Errorf("failed to create socket pair: %v", err)
+	}
+	defer socks[0].Close()
+	defer socks[1].Close()
+
+	// Get the underlying file for the second socket
+	sock1File, err := socks[1].(*net.UnixConn).File()
+	if err != nil {
+		return 0, fmt.Errorf("failed to get socket file: %v", err)
+	}
+	defer sock1File.Close()
+	// Prepare nsenter command to run fusermount in container's namespace
+	nsenterArgs := []string{
+		"--mount=" + nsPath,
+		s.fusermountPath,
+	}
+	nsenterArgs = append(nsenterArgs, req.Args...)
+
+	cmd := exec.Command("nsenter", nsenterArgs...)
+
+	// Pass the socket FD to fusermount
+	cmd.ExtraFiles = []*os.File{sock1File}
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("%s=3", common.EnvFuseCommFD), // FD 3 is the first extra file
+	)
+	// Capture command output for error reporting
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return 0, fmt.Errorf("fusermount failed: %v, output: %s", err, string(output))
+	}
+	// For mount operations, we need to transfer the fd back to the caller
+	if !strings.Contains(strings.Join(req.Args, " "), "-u") {
+		fd, _, err := mfcputil.RecvMsg(socks[0])
+		if err != nil {
+			return 0, fmt.Errorf("failed to receive FD: %v", err)
+		}
+		return fd, nil
+	}
+	return 0, nil
+}
+
+// ensureFuseDevice ensures that /dev/fuse exists in the container's namespace
+func (s *Server) ensureFuseDevice(nspath string) error {
+	// First check if /dev/fuse exists in the container's namespace
+	checkCmd := exec.Command("nsenter", "--mount="+nspath, "stat", "/dev/fuse")
+	if err := checkCmd.Run(); err == nil {
+		// Device exists, no need to create
+		return nil
+	}
+
+	// Device doesn't exist, create it
+	// Note: We need to create the device with mknod in the container's namespace
+	createCmd := exec.Command("nsenter",
+		"--mount="+nspath,
+		"mknod",
+		"/dev/fuse",
+		"c",
+		fmt.Sprintf("%d", FuseMajor),
+		fmt.Sprintf("%d", FuseMinor),
+	)
+
+	if output, err := createCmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to create /dev/fuse: %v, output: %s", err, string(output))
+	}
+
+	// Set permissions to allow user access
+	chmodCmd := exec.Command("nsenter",
+		"--mount="+nspath,
+		"chmod",
+		"666",
+		"/dev/fuse",
+	)
+
+	if output, err := chmodCmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to set permissions on /dev/fuse: %v, output: %s", err, string(output))
```

Comment:
```suggestion
		return fmt.Errorf("failed to set permissions on /dev/fuse: %w, output: %s", err, string(output))
```
to wrap errors in fmt.Errorf, see https://go.dev/blog/go1.13-errors

Reply:
Thanks!